aptos-move/aptos-aggregator/src/resolver.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::aggregator_extension::AggregatorID;
+
+/// Returns a value of an aggregator from cache or global storage.
+///   - Ok(..)       if aggregator value exists
+///   - Err(..)      otherwise.
+pub trait AggregatorResolver {
+    fn resolve_aggregator_value(&self, id: &AggregatorID) -> Result<u128, anyhow::Error>;
+}
+
+// Utils to store aggregator values in data store. Here, we
+// only care about aggregators which are state items.
+#[cfg(any(test, feature = "testing"))]
+pub mod test_utils {
+    use super::*;
+    use crate::{
+        aggregator_extension::AggregatorHandle,
+        delta_change_set::{deserialize, serialize},
+    };
+    use aptos_state_view::TStateView;
+    use aptos_types::state_store::{
+        state_key::StateKey, state_storage_usage::StateStorageUsage, state_value::StateValue,
+    };
+    use move_core_types::account_address::AccountAddress;
+    use move_table_extension::TableHandle;
+    use std::collections::HashMap;
+
+    /// Generates a dummy id for aggregator based on the given key. Only used for testing.
+    pub fn aggregator_id_for_test(key: u128) -> AggregatorID {
```

Comment:
Do we need this anymore? more precisely:
(a) should we be using legacy in testing anyway, since we hopefully will eventually migrate and try to get rid of it.
(b) if we use ephemeral for testing, it's much easier to convert a key, could maybe create more directly (depending on creation APIs)

Reply:
My plan was to add tests for ephemeral once they are added (right now all code is unreachable). Once we connect with other diffs, we can add a new testing flow, and when table items are gone (although there is only one😂) we can have 100% ephemeral ids usage.